features/integration/applications.feature

```diff
@@ -84,3 +85,131 @@ Feature: Applications
+
+      And I sleep for 500 milliseconds for indexer to digest things down.
+      And according to "indexer", the current application should have the following boxes "Zm9vIGJhcg==:APj/IA==:bmFtZQ==:MTE0NTE0".
+      And according to "indexer", with 2 being the parameter that limits results, the current application should have 2 boxes.
+      And according to "indexer", with 6 being the parameter that limits results, the current application should have 4 boxes.
+
+      # delete one box
+      Then I build an application transaction with the transient account, the current application, suggested params, operation "call", approval-program "", clear-program "", global-bytes 0, global-ints 0, local-bytes 0, local-ints 0, app-args "str:delete,str:name", foreign-apps "", foreign-assets "", app-accounts "", extra-pages 0, boxes "0,str:name"
+      And I sign and submit the transaction, saving the txid. If there is an error it is "".
+      And I wait for the transaction to be confirmed.
+      And according to "algod", the current application should have the following boxes "Zm9vIGJhcg==:APj/IA==:MTE0NTE0".
+
+      # move to indexer testing steps
+      And I sleep for 500 milliseconds for indexer to digest things down.
+      And according to "indexer", the contents of the box with name "str:name" in the current application should be "". If there is an error it is "no application boxes found".
+      And according to "indexer", the contents of the box with name "b64:APj/IA==" in the current application should be "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA". If there is an error it is "".
+      And according to "indexer", the current application should have the following boxes "Zm9vIGJhcg==:APj/IA==:MTE0NTE0".
+      And according to "indexer", with 2 being the parameter that limits results, the current application should have 2 boxes.
+      And according to "indexer", with 0 being the parameter that limits results, the current application should have 3 boxes.
+      And according to indexer, with 1 being the parameter that limits results, and "b64:APj/IA==" being the parameter that sets the next result, the current application should have the following boxes "MTE0NTE0".
+
+      # To test *exactly* which boxes are under an app with parameter `limit` can only be done through indexer,
+      # for indexer returns deterministic results with `ORDER BY`, but algod result on boxes are determined by box operations' order.
+      # To minimize the potential error space, the exact box comparison can be done only with indexer
+
+      # A potential improvement space:
+      # we can make this test determinstic (i.e., removing sleep for indexer) by waiting indexer to be at the same round of algod finishing all the box operations.
+      # Once indexer has a round number at least as large as algod finishing box operations, the changes on boxes can be reflected in indexer side.
```

Comment:
nit - this potential improvement may warrant a new issue as opposed to an inline comment.

Reply:
closing in favor of #238